src/app/components/work-item-iteration-modal/work-item-iteration-modal.component.ts

```diff
@@ -200,17 +207,28 @@ export class FabPlannerAssociateIterationModalComponent {
         this.selectedIteration = null;
         this.workItem.relationships.iteration = workItem.relationships.iteration;
         this.workItem.attributes['version'] = workItem.attributes['version'];
-        this.workItemService.emitEditWI(workItem);
+        this.workItemService.emitEditWI(this.workItem);
       });
   }
 
   open(event: any) {
     this.getIterations();
+    this.workItemPayload = {
```

Comment:
Why are you initiating payload in this function? workItem is an input to this component, should'd the payload be updated in **ngOnChanges** lifecycle hook? You'd want to update the payload as soon as the input changes with a work item.

Reply:
It is modal so, it only required when user open the modal. So I don't think I have to update it in ngOnchange.

Reply:
But in future, if you have to implement something case where while the modal is open, you need switch the work item and get the modal updated too, how would you do that? Life cycle hooks are there for a reason, and we should make use of it. Think about it, and even then you think it's okay to keep it here, I am good with that for now.

Reply:
Changes in https://github.com/fabric8-ui/fabric8-planner/pull/2170/commits/28b5a9ec89bbd4f81ec7e93fcf13eb175359d18d